### PR TITLE
ChangeFeedProcessorParsingLSNAsLongType

### PR DESCRIPTION
**Pull request by xinlian12**
There are few places within the changeFeedProcessor still parsing LSN as long type instead of int

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java [hunk 1]
```diff
@@ -98,8 +98,8 @@ public int getEstimatedLag() {
      * @param estimatedLag the estimated lag.
      * @return the current ChangeFeedProcessorState instance.
      */
-    public ChangeFeedProcessorState setEstimatedLag(int estimatedLag) {
-        this.estimatedLag = estimatedLag;
+    public ChangeFeedProcessorState setEstimatedLag(long estimatedLag) {
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java:101]: This is a breaking change, isn't it?

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
Change in public API is breaking - don't think that is something we can do - and I think it is fine to cap the estimated lag at int.maxValue - when using estimator it doesn't really matter anymore when there are more than int.maxValue transactions/lsns gap.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java:101]: yea agree - discussed with Kushagra earlier and agree that on public API level we should keep the same, and no need for breaking change. I will converted the change back in the next iteration iteration

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java]: updated

**Review by FabianMeiswinkel — APPROVED** [review]
THX - LGTM!

**Pull request merged**